### PR TITLE
Integrate `xcfilelist` input/output paths for script phases.

### PR DESCRIPTION
**Pull request by dnkoutso**
Needs https://github.com/CocoaPods/Core/pull/517

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Add support for `xcfilelist` in `script_phase` DSL  
  [dnkoutso](https://github.com/dnkoutso)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dnkoutso** [comment]
Needs changelog

File: lib/cocoapods/installer/user_project_integrator/target_integrator.rb [hunk 1]
```diff
@@ -233,6 +233,8 @@ def create_or_update_user_script_phases(script_phases, native_target)
               phase.shell_path = script_phase[:shell_path] || '/bin/sh'
               phase.input_paths = script_phase[:input_files]
               phase.output_paths = script_phase[:output_files]
+              phase.input_file_list_paths = script_phase[:input_file_lists]
+              phase.output_file_list_paths = script_phase[:output_file_lists]
```
Comment (amorde) [inline lib/cocoapods/installer/user_project_integrator/target_integrator.rb]: should there be a class/model for the script phase? seems like it's getting big enough to warrant one

Comment (dnkoutso) [reply lib/cocoapods/installer/user_project_integrator/target_integrator.rb]: probably...do we want it for this PR?

Comment (dnkoutso) [reply lib/cocoapods/installer/user_project_integrator/target_integrator.rb]: Also what would it give us? There is no state to manage

Comment (amorde) [reply lib/cocoapods/installer/user_project_integrator/target_integrator.rb]: better type safety, generated docs, better IDE integration, etc.

don't think it's necessary for this PR though

**Pull request merged**